JeongGod/3week/programmers/64064.py

```diff
@@ -0,0 +1,25 @@
+import re
+from itertools import permutations
+def solution(user_id, banned_id):
+    answer = set()
+    
+    def check(per):
+        # 해당 경우의 수가 가능한지 체크
+        idx = 0
+        for _, user in per:
+            ban = banned_id[idx].replace("*", ".")
+            pattern = re.compile(f"^{ban}$")
+
+            if re.match(pattern, user) is None:
+                return False
```

Comment:
```suggestion
            pattern = re.compile(f"^{ban}$")

            if pattern.match(user) is None:
                return False
```
저도 좀 찾아봤는데 re.compile을 이미 설정하셨으면, re.match처럼 다시 re를 호출하는 형식보다 pattern.match처럼 설정한 걸 사용하는 방식이 성능적으로 좋다고 합니다.

Reply:
저도 배워갑니다!